src/randoop/sequence/Statement.java

```diff
@@ -67,11 +94,198 @@ public boolean equals(Object o) {
   @Override
   public int hashCode() {
     int retval = 5;
-    retval += 7 ^ statement.hashCode();
+    retval += 7 ^ operation.hashCode();
     for (int i = 0; i < inputs.size(); i++) {
       retval += 13 ^ inputs.get(i).index;
     }
     return retval;
   }
 
+  public Class<?> getOutputType() {
+    return operation.getOutputType();
+  }
+
+  public List<Class<?>> getInputTypes() {
+    return operation.getInputTypes();
+  }
+
+  public void appendCode(Variable variable, List<Variable> inputs, StringBuilder b) {
+    if (!isVoidMethodCall()) {
+      Class<?> type = operation.getOutputType();
+      String typeName = type.getCanonicalName();
+      b.append(typeName);
+      b.append(" " + Variable.classToVariableName(type) + variable.index + " = ");
+    }
+    operation.appendCode(inputs, b);
+    b.append(";" + Globals.lineSep);
+  }
+
+  
+  public String toParseableString(String variableName, List<Variable> inputs) {
```

Comment:
Please add Javadoc about how this is intended to be used, especially since it is not an overriding definition.